### PR TITLE
Implement Header annotation

### LINKED ISSUE #487: [Service Based Implementation] Implement Header annotation
**Description:**
$Subject.  for the users to retrieve Headers from an http request. Refer to 2.3.4.5 section for the spec.  including optionals

```ballerina
resource function post hello1(@http:Header string referer) {
    
}

//Multiple header value extraction
resource function post hello2(@http:Header {name: "Accept"} string[] accept) {
    
}

public type RateLimitHeaders record {|
    string x\\-rate\\-limit\\-id;
    int x\\-rate\\-limit\\-remaining;
    string[] x\\-rate\\-limit\\-types;
|};

//Populate selected headers to a record
resource function get hello3(@http:Header RateLimitHeaders rateLimitHeaders) {
}
```

**Describe your problem(s)**

**Describe your solution(s)**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by Thevakumar-Luheerathan**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: ballerina-tests/tests/test.bal [hunk 1]
```diff
@@ -4,6 +4,198 @@ import ballerina/lang.value;
...
+    string readString = check io:fileReadString(jsonFilePath);
+    string replacedString = regex:replaceAll(readString, "(FUNC_NAME)", "httpHeaderTest10");
+    json readJson = check value:fromJsonString(replacedString);
+    json resp = check clientEndpoint->post("/post-hello-httpHeaderTest10", readJson);
+    json expectedResp = {"Outputs":{"resp":{"statusCode":400, "body":"no header value found for 'Hos'", "headers":{"Content-Type":"text/plain"}}}, "Logs":[], "ReturnValue":null};
+    test:assertEquals(resp, expectedResp);
+}
+
+@test:Config {}
+function httpHeaderTest11() returns error? {
```
Comment (xlight05) [inline ballerina-tests/tests/test.bal:128]: shall we make the test names, ballerina function names related descriptive? this is good for maintainability as if one test fails, we could get an idea easily but looking at the test results without diving into the code

File: native/src/main/java/io/ballerina/stdlib/azure/functions/Constants.java [hunk 2]
```diff
@@ -22,49 +22,55 @@
  * {@code Constants} contains the public constants to be used.
  */
 public interface Constants {
-    String SLASH = "/";
+ String SLASH = "/";
 
-    String PACKAGE_ORG = "ballerinax";
-    String PACKAGE_NAME = "azure_functions";
-    String HTTP_PACKAGE_ORG = "ballerina";
-    String HTTP_PACKAGE_NAME = "http";
```
Comment (xlight05) [inline native/src/main/java/io/ballerina/stdlib/azure/functions/Constants.java:30]: Check if the line spacing is correct and the IDE is up to date with styling guidelines provided by wso2.

File: compiler-plugin/src/main/java/org/ballerinax/azurefunctions/AzureDiagnosticCodes.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (xlight05) [inline compiler-plugin/src/main/java/org/ballerinax/azurefunctions/AzureDiagnosticCodes.java:2]: Should be 2022. Please check other files as well

File: compiler-plugin/src/main/java/org/ballerinax/azurefunctions/AzureCompilerPlugin.java [hunk 4]
```diff
@@ -28,7 +28,7 @@
 public class AzureCompilerPlugin extends CompilerPlugin {
     @Override
     public void init(CompilerPluginContext pluginContext) {
-//        pluginContext.addCodeAnalyzer(new AzureFunctionsCodeAnalyzer());
+        pluginContext.addCodeAnalyzer(new AzureFunctionsCodeAnalyzer());
 //        pluginContext.addCodeGenerator(new AzureCodeGenerator());
```
Comment (xlight05) [inline compiler-plugin/src/main/java/org/ballerinax/azurefunctions/AzureCompilerPlugin.java:32]: we can remove this line

**Review by xlight05 — CHANGES_REQUESTED** [review]

File: ballerina-tests/tests/test.bal (hunk 1 above)
Comment (Thevakumar-Luheerathan) [reply ballerina-tests/tests/test.bal:128]: Sure.. I will change

**Review by xlight05 — APPROVED** [review]

**Pull request merged**